### PR TITLE
[0.6/dx11] Add Debug Names and More

### PR DESCRIPTION
**Pull request by cwfitzgerald**
Does some general cleanup, supports alpha-to-coverage, and adds debug names to various resoures.

Debug spans don't work right for an unknown reason, for investigation later.

### CODE REVIEW

File: src/backend/dx11/src/debug.rs [hunk 1]
```diff
@@ -84,11 +57,44 @@ pub fn debug_marker(context: &ComPtr<d3d11::ID3D11DeviceContext>, args: fmt::Arg
...
+    }
+    res
+}
+
+/// Set the debug name of a resource.
+///
+/// Must be ASCII.
+///
+/// SetPrivateData will copy the data internally so the data doesn't need to live.
+pub fn debug_name(resource: &d3d11::ID3D11DeviceChild, name: &str) {
```
Comment (kvark) [inline src/backend/dx11/src/debug.rs:82]: let's call those as actions, i.e. `set_debug_name`

File: src/backend/dx11/src/device.rs [hunk 2]
```diff
@@ -1519,10 +1538,15 @@ impl device::Device<Backend> for Device {
...
-                unordered_access_views.push(
-                    self.view_image_as_unordered_access(&view)
-                        .map_err(|_| device::BindError::WrongMemory)?,
-                );
+                let uav = self
+                    .view_image_as_unordered_access(&view)
+                    .map_err(|_| device::BindError::WrongMemory)?;
+
+                if let Some(ref mut name) = image.internal.debug_name {
+                    debug_name_with_suffix(&uav, name, &format!(" -- UAV Mip {}", mip));
```
Comment (kvark) [inline src/backend/dx11/src/device.rs:1546]: if we are going with `format!`, let's use the baseline `debug_name` for it. There is no point to pretend that `debug_name_with_suffix` helps here :)

File: src/backend/dx11/src/lib.rs [hunk 3]
```diff
@@ -20,7 +20,6 @@ of continuous descriptor ranges (per type, per shader stage).
 extern crate bitflags;
 #[macro_use]
 extern crate log;
-#[macro_use]
 extern crate winapi;
```
Comment (kvark) [inline src/backend/dx11/src/lib.rs:23]: no need for `extern crate` declarations in Rust2018 if you aren't importing macros

File: src/backend/dx11/src/debug.rs [hunk 4]
```diff
@@ -73,7 +47,6 @@ impl Drop for DebugScope {
     }
 }
 
-#[cfg(debug_assertions)]
 pub fn debug_marker(context: &ComPtr<d3d11::ID3D11DeviceContext>, args: fmt::Arguments) {
```
Comment (kvark) [inline src/backend/dx11/src/debug.rs:50]: this should probably just get an `&str`. No point to receive the arguments if all we do is just creating a `String` for them, anyway

**Review by kvark — COMMENTED** [review]
A few minor notes. Overall looks great!

**Comment by cwfitzgerald** [comment]
Comments should be addressed, I squashed the commits as keeping them separate was too difficult.

**Comment by kvark** [comment]
thanks!
bors r+

**Pull request merged**